backend/tests/controllers/userControllers.test.ts

```diff
@@ -184,21 +164,57 @@ describe("Users Controllers - GET /users/:userId", () => {
   });
 
   it("should respond with a 404 status code for user not found", async () => {
+    const userId = "nonexist";
+    req.params = { userId };
 
-    getUserByIdMock = (userId: string) => ({
-      get: async () => ({
-        exists: false,
-        id: userId,
-        data: () => ({ name: "" }),
-      }),
-    });
-  
-  await getUserById(req, res, next);
-
-  expect(next).toHaveBeenCalledWith(
-    createError(404, "User not found.")
-  );
+    await getUserById(req, res, next);
 
+    // Assert that next is called with a 404 error
+    expect(next).toHaveBeenCalledWith(createError(404, "User not found."));
   });
 
 });
+
+describe("User Controller - PUT /users/:userId", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    req = {
+      params: {},
+      body: {},
+    } as unknown as Request;
+
+    res = {
+      status: jest.fn().mockReturnThis(),
+      json: jest.fn(),
+    } as unknown as Response;
+
+    next = jest.fn();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should delete a user in Firebase with valid data", async () => {
+    req.params.userId = "user1";
+
+    await deleteUser(req, res, next);
+
+    // Assert the response status and JSON
+    expect(res.status).toHaveBeenCalledWith(204);
+    expect(res.json).toHaveBeenCalledWith(
+      expect.objectContaining({ id: "user1"})
+    );
+  });
+
+  it("should respond with 404 status for non-existent user", async () => {
+    req.params.userId = "nonexistent";
+    req.body = { name: "deleted Name" };
+
+    await deleteUser(req, res, next);
```

Comment:
I do not understand how are you controlling the behavior of the deletFirebase Mock, can you explain where are you setting up if a delete call to firebase will resolve in an exception or a proper response?

Reply:
In the `getUserByIdMock` function, I've set up a mock implementation for the `delete` method. This implementation returns a resolved Promise, ensuring that when I call the `delete` method, it will consistently resolve without encountering any errors. This is why you'll notice the line where I use `return Promise.resolve();`.

Reply:
Got it, this approach only validate calls to the function but it is not validating params neither the calls that shouldn't been call like creat method, please add validations for the methods that shouldn't been call on a Delete operation